apps/meteor/client/views/root/SAMLLoginRoute.tsx

```diff
@@ -38,20 +38,6 @@ const SAMLLoginRoute = () => {
 		});
 	}, [dispatchToastMessage, rootUrl, router]);
 
-	const userId = useUserId();
-	useEffect(() => {
-		if (!userId) {
-			return;
-		}
-
-		router.navigate(
-			{
-				pathname: '/home',
-			},
-			{ replace: true },
-		);
-	}, [userId, router]);
-
```

Comment:
Just trying to understand: why are we removing this?
Shouldn't we keep redirecting the user to the homepage in case they are already logged in RC when accessing this page?

Reply:
Pierre had done a workaround, for some reason, in Meteor 3.0.0 this callback would never be called. However, in Meteor 2.0.0 it happens. 

If you stop to think about it, it doesn't make sense to redirect the user after logging in if he doesn't have a uid, because that means it failed.

Likewise, it doesn't make sense to redirect twice if it works.

You even did a test to see if it redirects to the home page twice, that's going to be quite interesting.